test/server/test_get_video.js

```diff
@@ -197,8 +204,63 @@ describe("Obtener video", () => {
     });
     video.save().then(() => {
       server.get(`/video/?usuario_id=1`).end((err, res) => {
-        expect(res).to.have.status(200);
-        expect(res.body.length).to.eq(1);
+        expect(res.body["videos"].length).to.eq(1);
+        expect(res.body["total"]).to.eq(1);
+        done();
+      });
+    });
+  });
+
+  it("get video de un usuario con video privado sin ser amigo devuelve vacio", (done) => {
+    const video = new Video({
+      url: "url/test",
+      titulo: "video test",
+      usuario_id: 1,
+      duracion: 60,
+      habilitado: true,
+      visibilidad: "privado",
+    });
+    video.save().then(() => {
+      server.get(`/video/?usuario_id=1&contactos[]`).end((err, res) => {
+        expect(res.body["videos"].length).to.eq(0);
+        expect(res.body["total"]).to.eq(0);
+        done();
+      });
+    });
+  });
+
+  it("get video de un usuario con video privado siendo amigo devuelve el video", (done) => {
+    const video = new Video({
+      url: "url/test",
+      titulo: "video test",
+      usuario_id: 1,
+      duracion: 60,
+      habilitado: true,
+      visibilidad: "privado",
+    });
+    video.save().then(() => {
+      server.get(`/video/?usuario_id=1&contactos[]=1`).end((err, res) => {
+        expect(res.body["videos"].length).to.eq(1);
+        expect(res.body["total"]).to.eq(1);
+        done();
+      });
+    });
+  });
+
+  it("get video de un usuario con video privado e inhabilitado siendo amigo devuelve vacio", (done) => {
+    const video = new Video({
+      url: "url/test",
+      titulo: "video test",
+      usuario_id: 1,
+      duracion: 60,
+      habilitado: true,
+      visibilidad: "privado",
+      habilitado: false,
+    });
+    video.save().then(() => {
+      server.get(`/video/?usuario_id=1&contactos[]=1`).end((err, res) => {
```

Comment:
Me hace ruido `contactos[]=1`, como sería se tengo varios contactos: `contactos[]=1,2,3` ?

Reply:
con esa nomenclatura seria contactos[]=1&contactos[]=2 pero es pasarle unaa lista de contactos. Por lo que lei hay varias formas de hacerlo, es mas, python lo hace de otra forma y node lo entiende de igual manera

Reply:
no quedaría más legible: `contactos=[]` si no tiene contactos y  `contactos=[1,2,3]` si tiene contactos?

Reply:
A pesar de que no me parece muy dificil de entender la query puntualmente si pongo eso que propones el parser de node rompe. La verdad que al ser una prueba no le di mayor relevancia y buscando encontre esa forma de pasar los arrays
https://www.apharmony.com/software-sagacity/2014/10/node-js-passing-arrays-in-the-querystring/